django/db/models/fields/related.py

```diff
@@ -119,10 +119,18 @@ def _check_related_name_is_valid(self):
         import re
         import keyword
         related_name = self.remote_field.related_name
-
-        is_valid_id = (related_name and re.match('^[a-zA-Z_][a-zA-Z0-9_]*$', related_name)
-                       and not keyword.iskeyword(related_name))
-        if related_name and not (is_valid_id or related_name.endswith('+')):
+        if not related_name:
+            return []
+        is_valid_id = True
+        if keyword.iskeyword(related_name):
+            is_valid_id = False
+        try:
```

Comment:
How about adding comments at the end of `try:` and `except:` lines?
I think it is better than `six.PY3` because of EAFP principle